pkg/inject/inject.go

```diff
@@ -281,13 +270,11 @@ func (conf *ResourceConfig) GetPodPatch(injectProxy bool) ([]byte, error) {
 		}
 	}
 
-	patch := &podPatch{
-		Values:      *values,
-		Annotations: map[string]string{},
-		Labels:      map[string]string{},
-	}
+	patch := newPodPatch(*values)
+
 	switch strings.ToLower(conf.workload.metaType.Kind) {
 	case k8s.Pod:
+		patch.addRemovals(conf.pod)
```

Comment:
Good suggestion about renaming that function :+1: 
Not adding the remove statements because the pod hasn't been injected is a normal case, so I wouldn't return an error, which I prefer to reserve to an exceptional event. It does makes sense to me though to log that case, right before `addRemovals` bails out :+1: